src/helix/users.rs

```diff
@@ -284,39 +290,56 @@ pub mod create_user_follows {
     impl helix::Request for CreateUserFollowsRequest {
         type Response = CreateUserFollows;
 
+        #[cfg(feature = "client")]
         const OPT_SCOPE: &'static [twitch_oauth2::Scope] = &[];
         const PATH: &'static str = "users/follows";
+        #[cfg(feature = "client")]
         const SCOPE: &'static [twitch_oauth2::Scope] = &[twitch_oauth2::Scope::UserEditFollows];
     }
 
     impl helix::RequestPost for CreateUserFollowsRequest {
         type Body = CreateUserFollowsBody;
 
-        fn result<RE>(
+        fn parse_response(
             self,
-            url: &url::Url,
-            body: &str,
+            uri: &http::Uri,
             response: http::Response<Vec<u8>>,
         ) -> Result<
             helix::Response<Self, <Self as helix::Request>::Response>,
-            helix::RequestError<RE>,
+            helix::HelixRequestPostError,
         >
         where
-            RE: std::error::Error + Send + Sync + 'static,
             Self: Sized,
         {
+            let text = std::str::from_utf8(&response.body())
+                .map_err(|e| helix::HelixRequestPostError::Utf8Error(response.body().clone(), e))?;
+            if let Ok(helix::HelixRequestError {
+                error,
+                status,
+                message,
+            }) = serde_json::from_str::<helix::HelixRequestError>(&text)
+            {
+                return Err(helix::HelixRequestPostError::Error {
+                    error,
+                    status: status.try_into().unwrap_or(http::StatusCode::BAD_REQUEST),
+                    message,
+                    uri: uri.clone(),
+                    body: response.body().clone(),
+                });
+            }
+
             let response = response.status().try_into().map_err(|_| {
                 // This path should never be taken, but just to be sure we do this
-                helix::RequestError::HelixRequestPostError {
+                helix::HelixRequestPostError::Error {
                     status: response.status(),
-                    url: url.clone(),
-                    body: body.to_string(),
+                    uri: uri.clone(),
+                    body: response.body().clone(),
                     message: String::new(), // FIXME: None, but this branch should really never be hit
                     error: String::new(),
                 }
             })?;
             Ok(helix::Response {
-                data: vec![response],
+                data: vec![response], // FIXME: This should be a bit better...
```

Comment:
I think I want to make response specify what the returned data is. However there's only a handful of endpoints that can only return one set of results

Reply:
some endpoints also don't return `helix::Response` only a status code

Reply:
some endpoints also don't return `helix::Response` and only a statuscode. 

https://dev.twitch.tv/docs/api/reference#delete-user-follows for example